core/src/lib.rs

```diff
@@ -36,6 +36,7 @@
 //! ```
 
 #![deny(clippy::all, unused_must_use, unused_imports)]
+#![feature(or_patterns)]
```

Comment:
this isn't really necessary but it's nice. I checked the rustfmt features that we're still on nightly for, and it doesn't seem like  they're getting stablized any time soon. If we want to switch to stable at some point, it wouldn't be hard to remove this feature.